pkg/client/record/event.go

```diff
@@ -153,12 +154,11 @@ func recordToSink(sink EventSink, event *api.Event, eventCorrelator *EventCorrel
 
 func isKeyNotFoundError(err error) bool {
 	statusErr, _ := err.(*errors.StatusError)
-	// At the moment the server is returning 500 instead of a more specific
-	// error. When changing this remember that it should be backward compatible
-	// with old api servers that may be still returning 500.
```

Comment:
It seems to have been returning a 404 for a long time (`generic/etcd/etcd.go` line 272). I'm still a bit fuzzy on how resources are handled by the apiserver, but it doesn't look like it should ever return a 500 for a missing event.